docs/data-sources/roleprivilege.md

```diff
@@ -0,0 +1,109 @@
+---
+# Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```

Comment:
Copyright year should be 2024

Reply:
@NavisJ I changed the copyright year of new go files to 2024, but changing the copyright year of doc file will need to modify the doc template. Directly change the doc will fail the "Terraform-main-CI / Generate". We can note it down for now and change it later for all new features.